composer/models/huggingface.py

```diff
@@ -82,65 +95,111 @@ def __init__(self,
                                                 conda_package='transformers',
                                                 conda_channel='conda-forge') from e
 
+        if peft_config is not None:
+            if not peft_installed:
+                raise MissingConditionalImportError(extra_deps_group='peft',
+                                                    conda_package='peft',
+                                                    conda_channel='conda-forge')
+
+        if peft_config is not None:
+            # Hugging Face requires the peft type and task type to be upper case, so we do that here
+            # https://github.com/huggingface/peft/blob/ebbff4023ad276cbcb2466fd7e99be7d3ae0ae11/src/peft/utils/peft_types.py#L22-L51
+            if isinstance(peft_config.peft_type, str):
+                peft_config.peft_type = peft_config.peft_type.upper()
+            if isinstance(peft_config.task_type, str):
+                peft_config.task_type = peft_config.task_type.upper()
+
+            if peft_config.peft_type != 'LORA':
+                raise ValueError(
+                    f'PEFT type {peft_config.peft_type} is not supported by HuggingFaceModel. Only LORA is supported.')
+
         super().__init__()
         self.model = model
-        self.config = model.config
-        self.model_forward_args = inspect.getfullargspec(self.model.forward).args
+        self.config: PretrainedConfig = model.config
+        self.model_forward_args = self._get_model_forward_args()
         self.tokenizer = tokenizer
+        self.should_filter_state_dict_peft = should_filter_state_dict_peft
+        self.use_logits = use_logits
+        self.labels: Optional[torch.Tensor] = None  # set in eval_forward() if exists
+        self.dummy_forward_called = False  # Used to make FSDP generate work, see generate function for more details
+        self.train_metrics: Optional[Dict] = self._get_metric_dict(metrics) if metrics is not None else None
+        self.val_metrics: Optional[Dict] = self._get_metric_dict(
+            eval_metrics) if eval_metrics is not None else copy.deepcopy(self.train_metrics)
+
+        is_causal_lm = _is_registered_causal_lm(self.model)
+        self.shift_labels = is_causal_lm if shift_labels is None else shift_labels
+
+        self._check_tokenizer_and_maybe_resize_embeddings(allow_embedding_resizing)
+
+        if is_causal_lm and not self.shift_labels:
+            log.warning('The shift_labels argument was set to False but the model is an instance of a'
+                        ' HuggingFace Causal LM. This may lead to incorrect behavior.')
+            # Note: No warning if shift_labels and not is_causal_lm, since the model may simply be a custom class.
+
+        if peft_config is not None:
+            self.model = _maybe_get_peft_model(peft_config, self.model)
 
+        self.using_peft = isinstance(self.model, PeftModel) if peft_installed else False
```

Comment:
nit
```suggestion
        self.is_peft_model = isinstance(self.model, PeftModel) if peft_installed else False
```

Reply:
I went with the current naming because `self` is not really a peft model, it contains a peft model. Would prefer to leave it, lmk if you object strongly.